src/test/java/fi/helsinki/opintoni/service/UserCleanerServiceTest.java

```diff
@@ -0,0 +1,145 @@
+
+    private static final String USER_TO_DELETE = "inactiveuser@helsinki.fi";
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private UserCleanerService userCleanerService;
+
+    @Autowired
+    private CalendarFeedRepository calendarFeedRepository;
+
+    @Autowired
+    private TodoItemRepository todoItemRepository;
+
+    @Autowired
+    private UsefulLinkRepository usefulLinkRepository;
+
+    @Autowired
+    private FavoriteRepository favoriteRepository;
+
+    @Autowired
+    private UserSettingsRepository userSettingsRepository;
+
+    @Autowired
+    private OfficeHoursRepository officeHoursRepository;
+
+    @Autowired
+    private ProfileRepository profileRepository;
+
+    @Autowired
+    private IAMClient iamClient;
+
+    private User getUserToBeDeleted() {
+        return userRepository.findByEduPersonPrincipalName(USER_TO_DELETE).orElse(null);
+    }
+
+    @Test
+    public void thatCalendarFeedIsDeleted() {
+        User user = getUserToBeDeleted();
+
+        assertThat(calendarFeedRepository.findByUserId(user.id).isPresent()).isTrue();
+        userCleanerService.cleanInactiveUsers();
+        assertThat(calendarFeedRepository.findByUserId(user.id).isPresent()).isFalse();
+    }
+
+    @Test
+    public void thatTodoItemsAreDeleted() {
+        User user = getUserToBeDeleted();
+
+        assertThat(todoItemRepository.findByUserId(user.id).isEmpty()).isFalse();
+        userCleanerService.cleanInactiveUsers();
+        assertThat(todoItemRepository.findByUserId(user.id).isEmpty()).isTrue();
+    }
+
+    @Test
+    public void thatUsefulLinksAreDeleted() {
+        User user = getUserToBeDeleted();
+
+        assertThat(usefulLinkRepository.findByUserIdOrderByOrderIndexAsc(user.id).isEmpty()).isFalse();
+        userCleanerService.cleanInactiveUsers();
+        assertThat(usefulLinkRepository.findByUserIdOrderByOrderIndexAsc(user.id).isEmpty()).isTrue();
+    }
+
+    @Test
+    public void thatFavoritesAreDeleted() {
+        User user = getUserToBeDeleted();
+
+        assertThat(favoriteRepository.findByUserIdOrderByOrderIndexAsc(user.id).isEmpty()).isFalse();
+        userCleanerService.cleanInactiveUsers();
+        assertThat(favoriteRepository.findByUserIdOrderByOrderIndexAsc(user.id).isEmpty()).isTrue();
+    }
+
+    @Test
+    public void thatOfficeHoursAreDeleted() {
+        User user = getUserToBeDeleted();
+
+        assertThat(officeHoursRepository.findByUserId(user.id).isEmpty()).isFalse();
+        userCleanerService.cleanInactiveUsers();
+        assertThat(officeHoursRepository.findByUserId(user.id).isEmpty()).isTrue();
+    }
+
+    @Test
+    public void thatOnlyInactiveUsersWithoutPortfolioAreDeletedFromUserAccountTable() {
+        int sizeBefore = userRepository.findAll().size();
+
+        userCleanerService.cleanInactiveUsers();
+        assertThat(userRepository.findAll().size()).isEqualTo(sizeBefore - 1);
```

Comment:
Tämä testihän ei löydä muita poistettavia käyttäjiä kuin tuon `inactiveuser`in (koska `IamService`n mockiin on kovakoodattu että vain ko. käyttäjä on poistokelpoinen), minkä takia koko testi on vähän turha kun servicen (huonosti nimetty) `hasActiveIamAccount`-tarkistusta ei suoriteta muille kuin `inactiveuser`ille (jolloin users-taulusta ei koskaan yritetäkään poistaa muita käyttäjiä).

Olisi parempi jos testissä mockattaisiin iam-kutsu palauttamaan `true` jollekin toisellekin käyttäjälle, jolloin testissä olisi enemmän järkeä.